src/main/kotlin/gov/nist/secauto/oscal/tools/server/OscalVerticle.kt

```
@@ -49,23 +50,196 @@ import gov.nist.secauto.oscal.tools.cli.core.CLI;
+            for (dir in paths) {
+                try {
+                    val expandedDir = expandHomeDirectory(dir.trim())
+                    val path = Paths.get(expandedDir).normalize().toAbsolutePath()
+                    
+                    // Validate the directory
+                    validateDirectory(path)
+                    
+                    // Add to allowed directories if validation passes
+                    allowedDirs.add(path)
+                    logger.info("Added allowed directory from OSCAL_SERVER_PATH: $path")
+                } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)